server/src/actions/storage.ts

```
@@ -1,121 +1,64 @@
+    try {
+      const accountName = req.body.accountName;
+      const accessKey = req.body.accessKey;
+      const endpointSuffix = req.body.endpointSuffix || 'core.windows.net';
+      const credentials = new SharedKeyCredential(accountName, accessKey);
+      const pipeline = StorageURL.newPipeline(credentials);
+      const serviceURL = new ServiceURL(`https://${accountName}.blob.${endpointSuffix}`, pipeline);
+      const aborter = Aborter.timeout(ONE_MINUTE);
+      let containerSegment = await serviceURL.listContainersSegment(aborter);
 
-    const aggregator = (err: Error, result: azure.BlobService.ListContainerResult, cb: any) => {
-      if (err) {
-        cb(err, []);
-      } else {
-        containers = containers.concat(result.entries);
-        if (result.continuationToken !== null) {
-          blobService.listContainersSegmented(result.continuationToken, aggregator);
-        } else {
-          cb(null, containers);
-        }
-      }
-    };
+      let containers = containerSegment.containerItems;
 
-    blobService.listContainersSegmented(continuationToken, (err: Error, result: azure.BlobService.ListContainerResult) => {
-      aggregator(err, result, (err: Error, containers: azure.BlobService.ContainerResult[]) => {
-        if (err) {
-          res.status(400).send(err);
-        } else {
-          res.status(200).send(containers);
-        }
-      });
-    });
+      while (containerSegment.nextMarker) {
+        containerSegment = await serviceURL.listContainersSegment(aborter, containerSegment.nextMarker);
+        containers = containers.concat(containerSegment.containerItems);
+      }
+      LogHelper.log('storage-containers', { containers: containers.length });
+      res.send(containers);
+    } catch (err) {
+      //This is mainly here to keep errors from crashing the server and will let us see what kinds of errors happen for the new server
+      LogHelper.error('storage-containers', err);
+      res.sendStatus(500);
```

will this throw a 500 for any 4xx errors? if so, can we return the 4xx errors?